src/App.tsx

```diff
@@ -2,72 +1,70 @@
-
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
-
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  const debounceDelay = 300;
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const applyQuery = useMemo(
+    () => debounce(setAppliedQuery, debounceDelay),
+    [],
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setSelectedPerson(null);
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  const handleSelectedPerson = useCallback((person: Person) => {
+    setSelectedPerson(person);
+  }, []);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
+  return (
+    <div className="container">
+      <main className="section is-flex is-flex-direction-column">
+        <h1 className="title" data-cy="title">
+          {selectedPerson
+            ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+            : 'No selected person'}
+        </h1>
+        <Dropdown
+          query={query}
+          onQueryChange={handleQueryChange}
+          filteredPeople={filteredPeople}
+          onSelected={handleSelectedPerson}
+          setQuery={setQuery}
```

Comment:
The prop `setQuery` should not be passed to the `Dropdown` component since direct DOM interactions should be avoided in favor of React state and lifecycle methods. If `setQuery` is intended to update the state in the parent component, consider lifting the state up or using a different approach that aligns with React best practices.

Reply:
don't pass `setQuery`.
use a separate `handle` function for it